nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -147,18 +151,19 @@ let
             ''
               mkdir $out
               diskImage=$out/disk.img
-              bootFlash=$out/bios.bin
               ${qemu}/bin/qemu-img create -f qcow2 $diskImage "40M"
               ${if cfg.useEFIBoot then ''
-                cp ${pkgs.OVMF-CSM.fd}/FV/OVMF.fd $bootFlash
-                chmod 0644 $bootFlash
+                efiVars=$out/EFI_VARS.fd
+                cp ${efiVars} $efiVars
+                chmod 0644 $efiVars
               '' else ''
               ''}
             '';
           buildInputs = [ pkgs.utillinux ];
-          QEMU_OPTS = if cfg.useEFIBoot
-                      then "-pflash $out/bios.bin -nographic -serial pty"
-                      else "-nographic -serial pty";
+          QEMU_OPTS = "-nographic -serial stdio -monitor none"
```

Comment:
Why `-monitor none`? I often rely on the monitor being available on stdio.

Reply:
Keep in mind that this is just for the boot disk / EFI vars generation. These `QEMU_OPTS` would not apply to the VM an end-user might use.

I switched `-serial` to `stdio` since when using `pty` there is no visible output in the nix logs. This makes debugging the script below much more difficult, since you can't see any error messages.

`-monitor` defaults to `stdio` which conflicts with `-serial stdio`, which is why I disabled it here.

Reply:
Right, makes sense! :)